### PR TITLE
experiment: simpler version of claudio/self-pass

### PR DESCRIPTION
**Pull request by crusso**
There's no need to add `self_id` to `ActorE` after all, thus avoiding much complexity.

The internal self declaration, if present, is enough.

The real culprit was some bad lowering that wasn't passing down the self-binding in the 
`ObjBlockE` case (now fixed).

### CODE REVIEW

File: src/lowering/desugar.ml [hunk 1]
```diff
@@ -90,8 +90,8 @@ and exp' at note = function
       (breakE "!" (nullE()))
       (* case ? v : *)
       (varP v) (varE v) ty).it
-  | S.ObjBlockE (s, _t, dfs) ->
-    obj_block at s None dfs note.Note.typ
+  | S.ObjBlockE (s, (self_id_opt,_), dfs) ->
```
Comment (crusso) [inline src/lowering/desugar.ml:93]: This was the bug, which was failing to introduce a the self declaration for some cases.

File: test/run-drun/ok/pass-class-self.diff-low.ok [hunk 2]
```diff
@@ -1,4 +1,5 @@
 --- pass-class-self.run
 +++ pass-class-self.run-low
-@@ -0,0 +1 @@
+@@ -0,0 +1,2 @@
 +Before!
++ys6dh-5cjiq-5dc
```
Comment (ggreif) [inline test/run-drun/ok/pass-class-self.diff-low.ok:5]: Why this diff? Didn't `run.ok` also change, did it?

Comment (crusso) [reply test/run-drun/ok/pass-class-self.diff-low.ok:5]: Good question...

Comment (crusso) [reply test/run-drun/ok/pass-class-self.diff-low.ok:5]: Hmm, why is the having any side-effect at all? The test is an actor class declaration and should do nothing until instantiated.

Comment (ggreif) [reply test/run-drun/ok/pass-class-self.diff-low.ok:5]: Yeah, it is probably doing what `drun` does, auto-deploying.

Comment (crusso) [reply test/run-drun/ok/pass-class-self.diff-low.ok:5]: Run doesn't instantiate the class, but run-ir etc do.

I tried this on master (removing the self references) and it seems to be existing behaviour. Somewhat odd though, but perhaps just an artifact of "compiled" code.

Comment (crusso) [reply test/run-drun/ok/pass-class-self.diff-low.ok:5]: Ah, the ir_interpreter decides to execute the body of an actor class with zero arguments (to match semantics of installation on ic).

Nvm.

Comment (ggreif) [reply test/run-drun/ok/pass-class-self.diff-low.ok:5]: It helps to see the side effects!

**Review by ggreif — APPROVED** [review]
LGTM! Nice improvement.

**Pull request merged**